### PR TITLE
Use strutil and randutil from x-go library

### PR DESCRIPTION
**Pull request by flotter**
Make Pebble depend on strutil and randutil packages supplied by the x-go library.

This patchset includes the following commits:

- Remove the strutil package from Pebble
- Move Pebble references to strutil.UUID() over to x-go randutil.RandomKernelUUID().
- Move Pebble references to strutil.MakeRandomString() over to x-go randutil.RandomString().
- Make Pebble timeutil package utilize x-go randutil.RandomDuration() instead of math/rand.

For more information on the x-go library, please visit https://github.com/canonical/x-go.

### CODE REVIEW

**Comment by anpep** [comment]
Nice work! Have a look at this feedback from mvo: https://github.com/canonical/x-go/pull/10#pullrequestreview-1077220459

**Comment by flotter** [comment]
**This PR has a dependency on x-go that is not yet in place!**

**Comment by flotter** [comment]
> **This PR has a dependency on x-go that is not yet in place!**

This is now resolved.

File: cmd/pebble/cmd_ls.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/jessevdk/go-flags"
 
 	"github.com/canonical/pebble/client"
-	"github.com/canonical/pebble/internal/strutil/quantity"
+	"github.com/canonical/x-go/strutil/quantity"
```
Comment (benhoyt) [inline cmd/pebble/cmd_ls.go:26]: Nit: this should now go in the group of import lines with `jessevdk/go-flags`. We normally have three import groups: stdlib, external, and project-internal (`canonical/pebble`).

Same in other files.

File: go.mod [hunk 2]
```diff
@@ -3,16 +3,14 @@ module github.com/canonical/pebble
...
-	github.com/kr/text v0.2.0 // indirect
 	github.com/pkg/term v1.1.0
 	golang.org/x/crypto v0.0.0-20201221181555-eec23a3978ad
-	golang.org/x/net v0.0.0-20210119194325-5f4716e94777 // indirect
 	golang.org/x/sys v0.0.0-20210119212857-b64e53b001e4
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c
 	gopkg.in/tomb.v2 v2.0.0-20161208151619-d5d1b5820637
-	gopkg.in/yaml.v2 v2.4.0
-	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
+	gopkg.in/yaml.v3 v3.0.1
```
Comment (benhoyt) [inline go.mod:15]: Was it intentional to update `yaml.v3` from v3.0.0 to v3.0.1? Sounds good to me, just curious.

**Review by benhoyt — APPROVED** [review]
Just one nit about import ordering, but looks good to me! 2652 lines deleted -- love it!

File: go.mod (hunk 2 above)
Comment (flotter) [reply go.mod:15]: Includes @niemeyer 's The Billion Dollar Mistake bugfix :)

File: cmd/pebble/cmd_ls.go (hunk 1 above)
Comment (flotter) [reply cmd/pebble/cmd_ls.go]: Fixed everywhere! Thank you.

**Comment by flotter** [comment]
> Just one nit about import ordering, but looks good to me! 2652 lines deleted -- love it!

Thanks for your review time!

**Review by niemeyer — APPROVED** [review]
Thanks for this, Fred!

There are some issues in randutil related to seeding, but I've mentioned these in x-go itself on a late review in canonical/x-go#17. I'm moving on with this PR here regardless as the fix should be in x-go itself.

**Pull request merged**